src/client/Microsoft.Identity.Client/ManagedIdentity/ManagedIdentityClient.cs

```diff
@@ -20,7 +20,7 @@ internal class ManagedIdentityClient
         private const string WindowsHimdsFilePath = "%Programfiles%\\AzureConnectedMachineAgent\\himds.exe";
         private const string LinuxHimdsFilePath = "/opt/azcmagent/bin/himds";
         private AbstractManagedIdentity identitySource;
-        public static ManagedIdentitySource s_sourceName = ManagedIdentitySource.None;
```

Comment:
I think the source name should remain static, because there is no guarantee that ppl will use 1 ManagedIdentityApplication

Reply:
fixed.